src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -84,173 +89,50 @@ const defaultProps = {
     priorityMode: CONST.PRIORITY_MODE.DEFAULT,
 };
 
-/**
- * @param {Object} nextUnreadReports
- * @param {Object} unreadReports
- * @returns {Boolean}
- */
-function checkForNewUnreadReports(nextUnreadReports, unreadReports) {
-    return nextUnreadReports.length > 0
-            && _.some(nextUnreadReports,
-                nextUnreadReport => !_.some(unreadReports, unreadReport => unreadReport.reportID === nextUnreadReport.reportID));
-}
-const memoizeCheckForNewUnreadReports = memoizeOne(checkForNewUnreadReports);
-
-/**
- * @param {Object} reportsObject
- * @returns {Array}
- */
-function getUnreadReports(reportsObject) {
-    const reports = _.values(reportsObject);
-    if (reports.length === 0) {
-        return [];
+class SidebarLinks extends React.Component {
+    constructor(props) {
+        super(props);
+        this.getRecentReportsOptionListItems = memoizeOne(this.getRecentReportsOptionListItems.bind(this));
```

Comment:
I'm not sure if we should keep `memoizeOne` since it just memoizes the last result, which seems kind of useless.

Reply:
Why do you think it's useless?

Reply:
Yeah, I hear ya. In [my second PR](https://github.com/Expensify/App/pull/10800) (which is based off this one) I have removed this in favor of no memoization.

Reply:
My understanding is that it caches the results once for _each set of unique arguments_.

```
foo(1) - not cached
foo(2) - not cached
foo(1) - cached
foo(1) - not cached
```

So, I don't think it's totally useless. It just allows each set of arguments to be cached once, so it cuts performance by up to 50%.

Reply:
>So if of the arguments change, which seems likely especially for activeReportID and reportActions, we would compute the results again.

Yes the point is to prevent re-computing them when they **do not change** because the returned result would be the same if the arguments have not changed. If they change then we want to recompute them.